legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/sqlQueryToString/dbSpecific/snowflake/snowflakeExtension.pure

```diff
@@ -234,3 +236,58 @@ function <<access.private>> meta::relational::functions::sqlQueryToString::snowf
+{
+  'CREATE LOCAL TEMP TABLE ' + $c.table.schema.database.name->toOne() + '.'+$c.table.schema.name->toOne() + '.' + $c.table.name->toOne() + '('+ $c.table.columns->cast(@Column).name->toOne() + ' ' + $c.table.columns->cast(@Column).type->toOne()->meta::relational::metamodel::datatype::dataTypeToSqlText() +');';
+}
+
+function meta::relational::functions::sqlQueryToString::snowflake::translateDropTableStatementForTempTable(d:DropTableSQL[1]): String[*]
+{
+  'DROP TABLE IF EXISTS ' + $d.table.schema.database.name->toOne() + '.' + $d.table.schema.name->toOne() + '.'+$d.table.name->toOne()+';';
+}
+
+function meta::relational::functions::sqlQueryToString::snowflake::translateLoadStatementForTempTable(l:LoadTableSQL[1], dbConfig: DbConfig[1]): String[*]
+{
+   let loadTempTableCommand = 'PUT file://'+$l.csvFileLocationVarPlaceHolder->toOne() + ' \'@'+$l.table.schema.database.name->toOne() + '.'+$l.table.schema.name->toOne()+'.%' + $l.table.name->toOne() +  '\' PARALLEL= 16 AUTO_COMPRESS=TRUE;';
+   let copyIntoCommand = 'COPY INTO '+$l.table.schema.database.name->toOne() + '.'+$l.table.schema.name->toOne() + '.' + $l.table.name->toOne() + ' FROM \'@'+$l.table.schema.database.name->toOne() + '.'+$l.table.schema.name->toOne() + '.%' + $l.table.name->toOne() + '\' file_format = (type= CSV field_optionally_enclosed_by = \'"\' skip_header=1);';
+   [$loadTempTableCommand, $copyIntoCommand];
+}
+
+function meta::relational::functions::sqlQueryToString::snowflake::translateCreateSchemaStatementForTempTable(c:CreateSchemaSQL[1]): String[*]
+{
+  'CREATE SCHEMA IF NOT EXISTS ' + $c.schema.database.name->toOne()+'.' + $c.schema.name->toOne() + ';';
+}
+
+function meta::relational::functions::sqlQueryToString::snowflake::translateUseDbStatementForTempTable(c:UseDbSQL[1]): String[*]
+{
+  let dbNameOnConnection = $c.dbOnConnection->toOne()->match([
+                                                              rc: meta::pure::alloy::connections::RelationalDatabaseConnection[1] | assert($rc.datasourceSpecification->cast(@meta::pure::alloy::connections::alloy::specification::SnowflakeDatasourceSpecification).databaseName->isNotEmpty(), 'databaseName on RelationalDatabaseConnection cannot be empty');
+                                                                                                                                    $rc.datasourceSpecification->cast(@meta::pure::alloy::connections::alloy::specification::SnowflakeDatasourceSpecification).databaseName;,
+                                                              c : DatabaseConnection[1]                                           | assert(false, 'Connection Type: DatabaseConnection is not Supported for useDb method');
+                                                                                                                                    '';
+                                                            ]);
+  'USE ' + $dbNameOnConnection + ';';
```

Comment:
Running the 'Use db' command would change the current user db. Will that not affect SQL execution which would have 'schema.table' references (with schema being present in the database mentioned on the connection)?